src/runtime/opengl.cpp

```diff
@@ -1930,6 +1930,28 @@ WEAK int halide_opengl_initialize_kernels(void *user_context, void **state_ptr,
     ModuleState **state = (ModuleState **)state_ptr;
     ModuleState *module = *state;
 
+    if (module) {
+        // Check to see if we've already compiled this set of kernels
+        // Currently, this check occurs by name.  We assume that if
+        // the *first* kernel was already compiled, then all of them
+        // were.
+
+        debug(user_context) << "Module already exists.\n";
+        const char* kernel_name = match_prefix(src, kernel_marker);
```

Comment:
How unique are kernel names? What if you have multiple Halide pipelines with different opengl stages that use the same Func name?